internal/app/machined/pkg/controllers/block/mount.go

```diff
@@ -0,0 +1,295 @@
+
+				// mount hasn't been done yet
+				if !ok {
+					var opts []mount.NewPointOption
+
+					// [TODO]: need to support more mount options:
+					// * proj quota (static)
+
+					opts = append(opts, mount.WithSelinuxLabel(volumeStatus.TypedSpec().MountSpec.SelinuxLabel))
+
+					if mountRequest.TypedSpec().ReadOnly {
+						opts = append(opts, mount.WithReadonly())
+					}
+
+					mountpoint := mount.NewPoint(
+						mountSource,
+						mountTarget,
+						mountFilesystem.String(),
+						opts...,
+					)
+
+					unmounter, err := mountpoint.Mount(mount.WithMountPrinter(logger.Sugar().Infof))
+					if err != nil {
+						return fmt.Errorf("failed to mount %q: %w", mountRequest.Metadata().ID(), err)
+					}
+
+					logger.Info("volume mount",
+						zap.String("volume", volumeStatus.Metadata().ID()),
+						zap.String("source", mountSource),
+						zap.String("target", mountTarget),
+						zap.Stringer("filesystem", mountFilesystem),
+					)
+
+					ctrl.activeMounts[mountRequest.Metadata().ID()] = mountContext{
+						point:     mountpoint,
+						readOnly:  mountRequest.TypedSpec().ReadOnly,
+						unmounter: unmounter,
+					}
+				} else if mountCtx.readOnly != mountRequest.TypedSpec().ReadOnly { // remount if needed
+					switch mountRequest.TypedSpec().ReadOnly {
+					case true:
+						err = mountCtx.point.RemountReadOnly()
+					case false:
+						err = mountCtx.point.RemountReadWrite()
+					}
+
+					if err != nil {
+						return fmt.Errorf("failed to remount %q: %w", mountRequest.Metadata().ID(), err)
+					}
+
+					mountCtx.readOnly = mountRequest.TypedSpec().ReadOnly
+				}
+
+				if err = safe.WriterModify(
+					ctx, r, block.NewMountStatus(block.NamespaceName, mountRequest.Metadata().ID()),
+					func(mountStatus *block.MountStatus) error {
+						mountStatus.TypedSpec().Spec = *mountRequest.TypedSpec()
+						mountStatus.TypedSpec().Source = mountSource
+						mountStatus.TypedSpec().Target = mountTarget
+						mountStatus.TypedSpec().Filesystem = mountFilesystem
+						mountStatus.TypedSpec().ReadOnly = mountRequest.TypedSpec().ReadOnly
+
+						return nil
+					},
+				); err != nil {
+					return fmt.Errorf("failed to create mount status %q: %w", mountRequest.Metadata().ID(), err)
+				}
+			}
+		}
+
+		r.ResetRestartBackoff()
+	}
+}
+
+func (ctrl *MountController) tearDownMountStatus(ctx context.Context, r controller.Runtime, logger *zap.Logger, mountRequest *block.MountRequest) (bool, error) {
+	logger = logger.With(zap.String("mount_request", mountRequest.Metadata().ID()))
+
+	okToDestroy, err := r.Teardown(ctx, block.NewMountStatus(block.NamespaceName, mountRequest.Metadata().ID()).Metadata())
+	if err != nil {
+		if state.IsNotFoundError(err) {
+			// no mount status, we are done
+			return true, nil
+		}
+
+		return false, fmt.Errorf("failed to teardown mount status %q: %w", mountRequest.Metadata().ID(), err)
+	}
+
+	if !okToDestroy {
+		logger.Warn("waiting for mount status to be torn down")
```

Comment:
Not a reason for a warning?
```suggestion
		logger.Info("waiting for mount status to be torn down")
```